src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/CompletionSource.cs

```diff
@@ -309,55 +421,53 @@ static void CheckForExperimentStatus(ITextView textView, in CompletionOptions op
                 itemsBuilder.Add(item);
             }
 
-            AddPropertiesToSession(session, completionList, triggerLocation);
-
             var filters = filterSet.GetFilterStatesInSet();
-            var items = itemsBuilder.ToImmutableAndFree();
+            var items = itemsBuilder.ToImmutable();
 
             if (completionList.SuggestionModeItem is null)
-                return new(items, suggestionItemOptions: null, selectionHint: AsyncCompletionData.InitialSelectionHint.RegularSelection, filters);
+                return (new(items, suggestionItemOptions: null, selectionHint: AsyncCompletionData.InitialSelectionHint.RegularSelection, filters), completionList);
 
             var suggestionItemOptions = new AsyncCompletionData.SuggestionItemOptions(
                 completionList.SuggestionModeItem.DisplayText,
                 completionList.SuggestionModeItem.Properties.TryGetValue(Description, out var description) ? description : string.Empty);
 
-            return new(items, suggestionItemOptions, selectionHint: AsyncCompletionData.InitialSelectionHint.SoftSelection, filters);
+            return (new(items, suggestionItemOptions, selectionHint: AsyncCompletionData.InitialSelectionHint.SoftSelection, filters), completionList);
+        }
 
-            static void AddPropertiesToSession(IAsyncCompletionSession session, CompletionList completionList, SnapshotPoint triggerLocation)
+        private static void AddPropertiesToSession(IAsyncCompletionSession session, CompletionList completionList, SnapshotPoint? triggerLocation)
+        {
+            // Store around the span this completion list applies to.  We'll use this later
+            // to pass this value in when we're committing a completion list item.
+            // It's OK to overwrite this value when expanded items are requested.
```

Comment:
1. why is it ok to overwrite later?
2. why do expanded items have a different span?

Reply:
question still applies.